tools/integration/src/index.ts

```diff
@@ -758,123 +817,359 @@ async function handleExport(argv: any) {
         })
     });
 
-    async function exportDashboard(dashboardId: string): Promise<any> {
-        try {
-            const url = `https://${server}/api/custom-dashboard/${dashboardId}`
-            logger.info(`Start to get the dashboard(id=${dashboardId}) from ${url}`);
+    const includes = Array.isArray(includeRaw) ? includeRaw : (includeRaw ? [includeRaw] : []);
+    const parsedIncludes = includes.map(item => {
+        const parts = parseIncludeItem(item);
+        const typePart = parts.find((p: string) => p.startsWith("type="));
+        const type = typePart ? typePart.split("=")[1] : "all";
+        const conditions = parts.filter((p: string) => !p.startsWith("type="));
+        if (!typePart) {
+            logger.warn(`'type=' missing in include clause "${item}", interpreting as type=all`);
+        }
+        return { type, conditions, explicitlyTyped: !!typePart };
+    });
 
-            const response = await axiosInstance.get(url, {
-                headers: {
-                    'Content-Type': 'application/json',
-                    'Authorization': `apiToken ${token}`
-                }
-            });
-            logger.info(`Successfully got the dashboard(id=${dashboardId}): ${response.status}`);
-            if (logger.isDebugEnabled()) {
-                logger.debug(`Response data: \n${JSON.stringify(response.data)}`);
+    const dashboardsPath = path.join(location, "dashboards");
+    const eventsPath = path.join(location, "events");
+    fs.mkdirSync(dashboardsPath, { recursive: true });
+    fs.mkdirSync(eventsPath, { recursive: true });
+
+    let wasDashboardFound = false;
+    let wasEventFound = false;
+
+    // Dashboard export
+    if (parsedIncludes.some(inc => inc.type === "dashboard" || inc.type === "all")) {
+        const allDashboards = await getDashboardList(server, token, axiosInstance);
+        let totalDashboardProcessed = 0;
+
+        for (const inc of parsedIncludes.filter(inc => inc.type === "dashboard" || inc.type === "all")) {
+            const matches = inc.conditions.filter(c => c.startsWith("id="));
+
+            let filtered;
+            if (matches.length && inc.explicitlyTyped) {
+                filtered = matches.map(idCond => {
+                    const id = idCond.split("=")[1]?.replace(/^"|"$/g, '');
+                    return { id };
+                });
+            } else if (matches.length) {
+                filtered = matches.map(idCond => {
+                    const id = idCond.split("=")[1]?.replace(/^"|"$/g, '');
+                    return allDashboards.find(d => d.id === id);
+                }).filter(Boolean);
+            } else {
```

Comment:
What are the difference between the two blocks? Why it doesn't have to check the existence of the id in allDashboards in the first if block?